### ISSUE TITLE
KtlintFormat Git Pre-Commit Hook stages entire file when changing any lines, even untouched hunks

### ISSUE THREAD

**Issue by Ciatzo** [JLLeitschuh/ktlint-gradle#451]
When running the ktLintFormat pre-commit hook, if format changes any line in a file, it stages the entire file. 

This happens in 
```
    echo "$CHANGED_FILES" | while read -r file; do
    if [ -f $file ]; then
        git add $file
    fi
```

when it adds all of `$file` instead of only the lines that were just changed by ktLintFormat.

When you have a file with an unused import, some staged changes and some unstaged changes, the hook removes the import and then promptly stages the entire file (including the purposefully unstaged changes).

**Comment by Ciatzo** [comment]
@Tapchicoma this seems like a bug right?

**Comment by Tapchicoma** [comment]
I would say it is rather enhancement to the git hook as described behaviour is intentional (easy to implement).

Feel free to contribute better git hook implementation.

**Comment by Ciatzo** [comment]
@Tapchicoma I'll see what i can do, you can assign me to this issue

**Comment by Tapchicoma** [comment]
Fixed via #470
